packages/core/strapi/lib/core/registries/components.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const { pickBy, has } = require('lodash/fp');
+const { createComponent } = require('../domain/component');
+const { hasNamespace, addNamespace } = require('../utils');
+
+const validateKeySameToSingularName = (components) => {
+  for (const cName of Object.keys(components)) {
+    const component = components[cName];
+
+    if (cName !== component.schema.info.singularName) {
+      throw new Error(
+        `The key of the component should be the same as its singularName. Found ${cName} and ${component.schema.info.singularName}.`
+      );
+    }
+  }
+};
+
+const componentsRegistry = () => {
```

Comment:
This registry was copied from content-types and adjusted comments and variables.